lib/main.js

```diff
@@ -50,6 +50,80 @@ const SEC_PKCS7EncoderOutputCallback = ctypes.FunctionType(ctypes.default_abi,
                                                             ctypes.char.ptr,
                                                             ctypes.int]);
 
+const PK11SlotInfo = ctypes.voidptr_t;
+// These structure definitions are needed,
+// because there is no function in NSS to get slot from certificate.
```

Comment:
Would using PK11_GetAllSlotsForCert work? I think that might make this simpler (you'll probably still have to declare a type for PK11SlotList and PK11SlotListElement).
For reference:
https://dxr.mozilla.org/mozilla-central/source/security/nss/lib/pk11wrap/pk11pub.h#851
https://dxr.mozilla.org/mozilla-central/source/security/nss/lib/pk11wrap/secmodt.h#99
https://dxr.mozilla.org/mozilla-central/source/security/nss/lib/pk11wrap/pk11pub.h#27